Test failure for WSL 1

Hi,

I tried to build KLEE and run tests on [WSL](https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux). All tests were either passed or expectedly failed except only one: _Runtime/POSIX/Futimesat.c_. 

> $ ":" "RUN: at line 4"
> $ "/home/yatcheniy/klee/build/bin/klee" "--output-dir=/home/yatcheniy/klee/build/test/Runtime/POSIX/Output/Futimesat.c.tmp.klee-out" "--posix-runtime" "--exit-on-error" "/home/yatcheniy/klee/build/test/Runtime/POSIX/Output/Futimesat.c.tmp2.bc" "--sym-files" "1" "10"
> # command stderr:
> KLEE: NOTE: Using POSIX model: /home/yatcheniy/klee/build/runtime/lib/libkleeRuntimePOSIX64_Debug+Asserts.bca
> KLEE: output directory is "/home/yatcheniy/klee/build/test/Runtime/POSIX/Output/Futimesat.c.tmp.klee-out"
> KLEE: Using Z3 solver backend
> KLEE: WARNING: undefined reference to function: __fgetc_unlocked
> KLEE: WARNING: undefined reference to function: __fputc_unlocked
> KLEE: WARNING: undefined reference to function: endutent
> KLEE: WARNING: undefined reference to function: getutent
> KLEE: WARNING: undefined reference to function: realpath
> KLEE: WARNING: undefined reference to function: setutent
> KLEE: WARNING: undefined reference to function: strcmp
> KLEE: WARNING: undefined reference to function: strlen
> KLEE: WARNING: undefined reference to function: strncmp
> KLEE: WARNING: undefined reference to function: utmpname
> KLEE: WARNING ONCE: Alignment of memory from call "malloc" is not modelled. Using alignment of 8.
> KLEE: WARNING ONCE: calling external: syscall(4, 71149608, 69864848) at runtime/POSIX/fd.c:544 12
> KLEE: WARNING ONCE: calling __klee_posix_wrapped_main with extra arguments.
> KLEE: WARNING ONCE: calling external: gettimeofday(71258000, 0) at runtime/POSIX/stubs.c:174 7
> KLEE: WARNING ONCE: calling external: strlen(68206592) at runtime/POSIX/fd.c:46 14
> KLEE: WARNING ONCE: calling external: strncmp(71161280, 68206592, 45) at runtime/POSIX/fd.c:51 43
> KLEE: ERROR: test/Runtime/POSIX/Futimesat.c:38: ASSERTION FAIL: r != -1
> KLEE: NOTE: now ignoring this error at this location
> KLEE: ERROR: EXITING ON ERROR:
> Error: ASSERTION FAIL: r != -1
> File: test/Runtime/POSIX/Futimesat.c
> Line: 38
> assembly.ll line: 1216
> State: 1
> Stack:
>         #000001216 in __klee_posix_wrapped_main (=1, =71256416) at test/Runtime/POSIX/Futimesat.c:38
>         #100001088 in main (=4, =69864416, =69864456) at runtime/POSIX/klee_init_env.c:245
> 
> 
> 
> error: command failed with exit status: 1
> 
> --
> 
> ********************
> ********************
> Failed Tests (1):
>   KLEE :: Runtime/POSIX/Futimesat.c

When I dug a little bit deeper, it turned out, that `syscall(__NR_futimesat, ...)` returns -1 and `errno` is set to `ENOSYS`. That means particular syscall is not supported on given version of WSL.

I would suggest to change the test case as follows. Firstly, check whether given syscall is supported or not by sending it directly. Secondly, change the checks as though it was concrete execution.

If you need extra information, please let me know. Also I am ready to provide redesigned test case soon, if that's acceptable for you.

Relevant information, that can help you:

- PS C:\> wsl -l -v
  NAME                       STATE             VERSION
  Ubuntu-18.04           Running         1
-  root@Yatcheniyhualap:/home/yatcheniy/klee# uname -r
4.4.0-19041-Microsoft
- commit 24badb5bf17ff586dc3f1856901f27210713b2ac
- The CXX compiler identification is Clang 10.0.0
- The C compiler identification is Clang 10.0.0

--- a/test/Runtime/POSIX/Futimesat.c
+++ b/test/Runtime/POSIX/Futimesat.c
@@ -1,3 +1,5 @@
+// REQUIRES: not-wsl-1
+// Disabling WSL 1 because futimesat syscall may be not implemented
 // RUN: %clang %s -emit-llvm %O0opt -g -c -DTDIR=%T -o %t2.bc
 // RUN: touch %T/futimesat-dummy
 // RUN: rm -rf %t.klee-out

--- a/test/lit.cfg
+++ b/test/lit.cfg
@@ -211,6 +211,14 @@ for target in supported_targets:
     config.available_features.add(target)
   else:
     config.available_features.add('not-{}'.format(target))
+if 'WSL_DISTRO_NAME' in os.environ and 'WSL_INTEROP' not in os.environ:
+  config.available_features.add('wsl-1')
+else:
+  config.available_features.add('not-wsl-1')
+if 'WSL_DISTRO_NAME' in os.environ and 'WSL_INTEROP' in os.environ:
+  config.available_features.add('wsl-2')
+else:
+  config.available_features.add('not-wsl-2')
 
 # Sanitizer
 config.available_features.add('{}asan'.format('' if config.have_asan else 'not-'))

